packages/twenty-front/src/modules/auth/hooks/useAuth.ts

```diff
@@ -126,13 +125,23 @@ export const useAuth = () => {
   );
 
   const handleSignOut = useCallback(() => {
-    goToRecoilSnapshot(snapshot_UNSTABLE());
     setTokenPair(null);
     setCurrentUser(null);
+    setCurrentWorkspaceMember(null);
```

Comment:
@a-kud we actually would like to keep removing all recoil states (table contents, favorites, ...) but keep the iconsState + clientConfigState.

can we try to transform handleSignOut into a useRecoilCallback() and to keep goToRecoilSnapshot(snapshot_UNSTABLE()) but get the content if iconsState and clientConfigState before (from snapshot to avoid re-renders, that's the point of using a useRecoilCallback) an re-set it after?

Reply:
can you elaborate, please?
should it look like something like this?
```const handleSignOut = useRecoilCallback(
  ({ snapshot, reset, set }) => async () => {

    const iconsValue = snapshot.getLoadable(iconsState).contents;
    const clientConfigValue = snapshot.getLoadable(clientConfigState).contents;

    reset(currentUserState);
    reset(currentWorkspaceMemberState);
    reset(currentWorkspaceState);
    reset(isVerifyPendingState);
    reset(tokenPairState);
    
    await client.clearStore();
    sessionStorage.clear();

    set(iconsState, iconsValue);
    set(clientConfigState, clientConfigValue);
  },
  [],
);```

Reply:
Yes, exactly!

Reply:
A clean way of doing it would to craft the initial snapshot we want by using .map() like this : 

```typescript
const emptySnapshot = snapshot_UNSTABLE()

const initialSnapshot = emptySnapshot.map(({ set }) => {
  set(iconsState, iconsValue);
  set(clientConfigState, clientConfigValue);
});

goToRecoilSnapshot(initialSnapshot)
```

Then to obtain `iconsValue` and `clientConfigValue` we can either use a useRecoilCallback that will just read those two states or directly using `useRecoilValue`.